### PR TITLE
netspeed: show all ipv6 addrs in device details dialog

### PR DESCRIPTION
**Pull request by rbuj**
Screenshot at 2020-12-18 20-12-18
Screenshot at 2020-12-18 20-12-47
Screenshot at 2020-12-18 20-13-25

### CODE REVIEW

**Review by vkareh — APPROVED** [review]
works well for me, thanks!

**Comment by vkareh** [comment]
Oh, just saw the latest update with the `global` and `link-local`, etc. Looks good to me as well :+1:

File: netspeed/src/backend.c [hunk 1]
```diff
@@ -105,6 +111,77 @@ get_available_devices (void)
...
+                void *tempAddrPtr = NULL;
+
+                if (ifa->ifa_addr->sa_family == AF_INET6) {
+                    const char *scope;
+                    guint8 prefix6[16];
+                    guint  word, shiff;
+                    guint  prefix_lenght = 0;
+                    gboolean check = TRUE;
+
+                    tempAddrPtr = &((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;
```
Comment (raveit65) [inline netspeed/src/backend.c:142]: Build warning.
```
backend.c:142:37: warning: cast from 'struct sockaddr *' to 'struct sockaddr_in6 *' increases required alignment from 2 to 4 [-Wcast-align]

                    tempAddrPtr = &((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;

                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750455224#L5932

File: netspeed/src/backend.c [hunk 2]
```diff
@@ -105,6 +111,77 @@ get_available_devices (void)
...
+                        scope = _("v4compat");
+                    } else if (IN6_IS_ADDR_LOOPBACK (tempAddrPtr)) {
+                        scope = _("host");
+                    } else if (IN6_IS_ADDR_UNSPECIFIED (tempAddrPtr)) {
+                        scope = _("unspecified");
+                    } else {
+                        scope = _("global");
+                    }
+                    inet_ntop (ifa->ifa_addr->sa_family, tempAddrPtr, ip, sizeof (ip));
+                    memcpy(prefix6, &((struct sockaddr_in6 *) ifa->ifa_netmask)->sin6_addr, 16);
```
Comment (raveit65) [inline netspeed/src/backend.c:159]: Same.
```
backend.c:159:39: warning: cast from 'struct sockaddr *' to 'struct sockaddr_in6 *' increases required alignment from 2 to 4 [-Wcast-align]

                    memcpy(prefix6, &((struct sockaddr_in6 *) ifa->ifa_netmask)->sin6_addr, 16);

                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750455224#L5935

File: netspeed/src/backend.c [hunk 3]
```diff
@@ -105,6 +111,77 @@ get_available_devices (void)
...
+                                break;
+                            prefix6[word] <<= 1;
+                            prefix_lenght++;
+                        }
+                    }
+                    list = g_slist_prepend (list,
+                                            g_strdup_printf ("%s: %s/%u",
+                                                             scope, ip, prefix_lenght));
+                } else {
+                    tempAddrPtr = &((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
```
Comment (raveit65) [inline netspeed/src/backend.c:172]: Another one.
```
backend.c:172:37: warning: cast from 'struct sockaddr *' to 'struct sockaddr_in *' increases required alignment from 2 to 4 [-Wcast-align]

                    tempAddrPtr = &((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;

                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750455224#L5938

File: netspeed/src/backend.c [hunk 4]
```diff
@@ -105,6 +111,77 @@ get_available_devices (void)
...
+                    } else if (IN6_IS_ADDR_UNSPECIFIED (tempAddrPtr)) {
+                        scope = _("unspecified");
+                    } else {
+                        scope = _("global");
+                    }
+                    inet_ntop (ifa->ifa_addr->sa_family, tempAddrPtr, ip, sizeof (ip));
+                    memcpy(prefix6, &((struct sockaddr_in6 *) ifa->ifa_netmask)->sin6_addr, 16);
+                    for (word = 0; word < 16; word ++) {
+                        for (shiff = 0; shiff < 8; shiff++) {
+                            if ((check = (prefix6[word] & (1<<7))) == FALSE)
```
Comment (raveit65) [inline netspeed/src/backend.c:162]:
```
backend.c:162:34: warning: Although the value stored to 'check' is used in the enclosing expression, the value is never actually read from 'check' [deadcode.DeadStores]

                            if ((check = (prefix6[word] & (1<<7))) == FALSE)

                                 ^       ~~~~~~~~~~~~~~~~~~~~~~~~
```
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750455224#L6039

**Comment by raveit65** [comment]
Beside from the build warnings the dialog displays the ip6 adresses.

**Comment by rbuj** [comment]
@raveit65 done

File: netspeed/src/backend.c (hunk 4 above)
Comment (raveit65) [reply netspeed/src/backend.c]: thx, warning is gone.

File: netspeed/src/backend.c (hunk 1 above)
Comment (raveit65) [reply netspeed/src/backend.c]: Hmm, warnings are still there.
https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750571886#L5929

Comment (rbuj) [reply netspeed/src/backend.c]: I can't see it on my local build:
```
$ CFLAGS="-Wcast-align" ./autogen.sh --prefix=/usr && make &> make.log
$ grep -F "Wcast-align" make.log 
```

**Comment by raveit65** [comment]
Just a rebase with master to trigger another travis build.

File: netspeed/src/backend.c (hunk 1 above)
Comment (raveit65) [reply netspeed/src/backend.c]: Confirmed, i don't get that warning when building local with  `-Wcast-align`, but in latest debian build from this branch it is there. https://travis-ci.org/github/mate-desktop/mate-applets/jobs/750599329#L4410
Any way, it's not reproducible for me.

**Pull request merged**